pandas/io/pickle.py

```diff
@@ -1,25 +1,38 @@
 """ pickle compat """
 import pickle
+from typing import Any, Optional
 import warnings
 
+from pandas._typing import FilePathOrBuffer
 from pandas.compat import pickle_compat as pc
 
-from pandas.io.common import get_handle, stringify_path
+from pandas.io.common import get_filepath_or_buffer, get_handle
 
 
-def to_pickle(obj, path, compression="infer", protocol=pickle.HIGHEST_PROTOCOL):
+def to_pickle(
+    obj: Any,
+    filepath_or_buffer: FilePathOrBuffer,
+    compression: Optional[str] = "infer",
+    protocol: int = pickle.HIGHEST_PROTOCOL,
+):
     """
     Pickle (serialize) object to file.
 
     Parameters
     ----------
     obj : any object
         Any python object.
-    path : str
-        File path where the pickled object will be stored.
+    filepath_or_buffer : str, path object or file-like object
+        File path, URL, or buffer where the pickled object will be stored.
+
+        .. versionchanged:: 1.0.0
+           Accept URL. URL has to be of S3 or GCS.
```

Comment:
Is the S3 or GCS restriction still accurate? You seem to have tests for general `http://` URLs.

Reply:
`get_filepath_or_buffer` handles URL only one way: [sending GET request to the URL using `urllib.request.urlopen()` (only arg it takes is the URL)](https://github.com/pandas-dev/pandas/blob/master/pandas/io/common.py#L171-L179).
I wrote a test, but it only test `read_pickle` for this reason.

Should I update `to_pikle` so that it'll send POST request...? 
I thought it might be a bit too much expansion. 
But please let me know, I'd be happy to work on it.

Just as a side note, for S3/GCS, we use their library and receive a buffer, so we can write just by changing the mode.

Reply:
Understood. We can handle that elsewhere.